conan/tools/cmake/toolchain/blocks.py

```diff
@@ -565,10 +565,10 @@ class ExtraFlagsBlock(Block):
 
     def context(self):
         # Now, it's time to get all the flags defined by the user
-        cxxflags = self._conanfile.conf.get("tools.build:cxxflags", default=[], check_type=list)
-        cflags = self._conanfile.conf.get("tools.build:cflags", default=[], check_type=list)
-        sharedlinkflags = self._conanfile.conf.get("tools.build:sharedlinkflags", default=[], check_type=list)
-        exelinkflags = self._conanfile.conf.get("tools.build:exelinkflags", default=[], check_type=list)
+        cxxflags = self._conanfile.conf.get("tools.build:cxxflags", default=[], check_type=list) + self._toolchain.extra_cxxflags
```

Comment:
cxxflags have higher priority if they are appended. I think that the profile flags should have priority over the recipe flags if there is conflict, so we might want to reverse this order, what do you think?

Reply:
Yes, totally makes sense, I'll change the preference order